### PR TITLE
fix(select): allow select to be shallowDisabled

### PR DESCRIPTION
**Pull request by chburket**
# Description

Added support for the `shallowDisabled` property that makes the Select "mostly" disabled but still allows it to receive focus.

### CODE REVIEW

File: src/components/Select/Select.tsx [hunk 1]
```diff
@@ -48,7 +49,10 @@ function Select<T extends object>(props: Props<T>, ref: RefObject<HTMLDivElement
   const state = useSelectState(props);
   const { labelProps, triggerProps, valueProps, menuProps } = useSelect(props, state, selectRef);
   const { buttonProps } = useButton({ ...triggerProps, isDisabled }, selectRef);
-  const ariaActivedesecendant = menuProps?.id && state?.selectionManager?.focusedKey && `${menuProps.id}-option-${state.selectionManager.focusedKey}`;
+  const ariaActivedesecendant =
+    menuProps?.id &&
+    state?.selectionManager?.focusedKey &&
+    `${menuProps.id}-option-${state.selectionManager.focusedKey}`;
```
Comment (chburket) [inline src/components/Select/Select.tsx:55]: Not sure why this reformatted. Are my prettier settings different than standard?

File: src/components/Select/Select.tsx [hunk 2]
```diff
@@ -181,15 +203,15 @@ function Select<T extends object>(props: Props<T>, ref: RefObject<HTMLDivElement
...
-          />
+        <ListBoxBase
+          {...menuProps}
+          ref={boxRef}
+          state={state}
+          disallowEmptySelection
+          // eslint-disable-next-line jsx-a11y/no-autofocus
+          autoFocus={state.focusStrategy || DEFAULTS.FOCUS_STRATEGY}
+          className={STYLE.menuListBox}
+        />
```
Comment (chburket) [inline src/components/Select/Select.tsx:214]: Not sure why this reformatted. Are my prettier settings different than standard?

File: src/components/Select/Select.tsx [hunk 3]
```diff
@@ -97,6 +101,24 @@ function Select<T extends object>(props: Props<T>, ref: RefObject<HTMLDivElement
     }
   }, []);
 
+  const otherProps = shallowDisabled
+    ? {
+        'aria-disabled': true,
+        'data-shallow-disabled': true,
```
[the remark this answers is not in the document]
Comment (chburket) [reply src/components/Select/Select.tsx:107]: This change will allow that.

[the remark this answers is not in the document]
Comment (jor-row) [reply src/components/Select/Select.tsx:107]: Cool, wasn't sure if that would be done in momentum or cantina

**Review by chrisadubois — APPROVED** [review]

**Pull request merged**